fmt.sh

```diff
@@ -105,18 +113,20 @@ echo "\
 # Check for correct copyrights
 run dade-copyright-headers "$dade_copyright_arg" .
 
-# We have a Bazel test that is meant to run HLint, but we're a little sceptical of it
-# If we get this far, but hlint fails, that's a problem we should fix
-function bad_hlint() {
-  echo "UNEXPECTED HLINT FAILURE: The Bazel rules should have spotted this, please raise a GitHub issue"
-}
-trap bad_hlint EXIT
-for dir in daml-assistant libs-haskell compiler release language-support; do
-  run pushd "$dir"
-  run hlint --git -j4
-  run popd
-done
-trap - EXIT
+if [ "$hlint_diff" = "true" ]; then
+    changed_haskell_files="$(git diff --name-only origin/master | grep '.hs$')"
+    if [ "" != "$changed_haskell_files" ]; then
+        hlint -j4 $changed_haskell_files
+    fi
+else
+    # We have a Bazel test that is meant to run HLint, but we're a little
```

Comment:
This comment might be a bit outdated by now. Afaik the bazel rules work just fine. The advantage of keeping the separate hlint calls in ./fmt.sh is that this way you get format errors at the very beginning.

Reply:
Happy to kill it. Should I?

Reply:
Yeah I would just kill it.